nncf/torch/quantization/init_range.py

```diff
@@ -115,10 +116,20 @@ def convert_reduction_shape(self, per_sample_stats) -> ReductionShape:
         if self._per_channel:
             val = (ndims + self._channel_idx) % ndims
             reduction_shape.remove(val)
+            if not val and self.use_per_sample_stats(per_sample_stats):
+                raise RuntimeError("Batch dimension should be equal to zero")
         if self.use_per_sample_stats(per_sample_stats):
             reduction_shape = reduction_shape[1:]  # Assumes batch is the first dimension
         return tuple(reduction_shape)
 
+    def convert_statistic_params(self, per_sample_stats):
```

Comment:
Ok, then make it better:
1) The original `convert_reduction_axes` wasn't named well, `get_reduction_axes` would be more fitting
2) Since this function basically returns reduction axes by calling the `get_reduction_axes` *and* something else too, you would do better to repurpose it to return only that something else, e.g. aggregation axes; then call it `get_aggregation_axes()` and then in the code where you need it do: 
```python
coll = TensorCollector(reduction_axes=range_init_params.get_reduction_axes(), aggregation_axes=range_init_params.get_aggregation_axes())
```

or whatever.

Reply:
Did it, please check